### DISCUSSION TITLE
ICMP / Ping Host Discovery / Manual Device Input

### DISCUSSION THREAD

**Discussion by No1234DDgg** [netalertx/NetAlertX#891] [Ideas]
First of all jakob-sk thanks a lot for this nice project :)

I would like to use NetAlertX as my asset inventory solution.

I have used enterprise software like Lansweeper and Spiceworks IT-Inventory and have to admit they are cool but so overloaded with features or cloud only.

Unfortunately I am not a developer otherwise I would help you out by trying my best to develop plugins :)

Is there a simple way to do host discovery just via icmp/ping? I have other networks that are offsite and not connected via layer2 so no ARP scan haha.

Also a simple via gui approach to adding new devices would be cool i guess.

Greetings from Germany

Thanks for your time 
-No1234DDgg

**Comment by jokob-sk** [comment]
I think the upcoming plugin (`IPNEIGH`) developed by @KayJay7 will be what you are looking for. 

I've also added an option to manually add devices in the next release. 

It would be great if you could test this. Can you please switch to the `netalertx-dev` docker image (backup everything at first or, ideally spin up a **separate container with a separate db and config**), in about 15 minutes (or after the last [action finishes](https://github.com/jokob-sk/NetAlertX/actions)) from now?

Make sure you refresh your browser cache - and click the 🔄 refresh button in the top right corner.

Thanks in advance,
j

**Comment by jokob-sk** [comment]
Realized that `IPNEIGH` only scans interfaces, so I created an `ICMP` plugin as well. Please test in ~15 min.

**Reply by No1234DDgg** [reply]
after the holidays i will do so :)

**Comment by EmuLabs** [comment]
Hi Dev, thank you for your efforts. ICMP plugin only scans known devices, how to get it to scan entire subnet to create new devices? i.e. scan 192.168.1.1-192.168.1.254.

**Reply by jokob-sk** [reply]
Hi @EmuLabs , 

The `ICMP` plugin only pings devices discovered by the app. These, in turn, are usually discovered via scans that target the IP ranges specified in the `SCAN_SUBNETS` setting.

The ICMP plugin can't be used for device discovery, because the only thing the ICMP scan does is to return if an IP is offline or online. Devices in NAX need to have at least a MAC address. Mac addresses are not returned by the ICMP utility as far as I know. 

That's why combining ICMP with other plugins is recommended, such as NMAPDEV or ARPSCAN.

**Reply by EmuLabs** [reply]
Hi Dev that makes perfect sense I don't know how I missed that. I appreciate you taking the time to explain that. 
Time for me to do some more tweaking!
